packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -340,3 +385,12 @@ function getAmiTypeForInstanceType(instanceType: InstanceType) {
           NodegroupAmiType.AL2_X86_64;
 }
 
+function determineAmiTypes(instanceType: InstanceType[]) {
+  const amiTypes = instanceType.map(i =>getAmiTypeForInstanceType(i));
+  const uniq = [...new Set(amiTypes)];
+  if (uniq.length > 1) {
+    throw new Error('instanceTypes of different CPU architectures not allowed');
```

Comment:
So is this effectively true from an EKS perspective as well? Making sure we are not introducing some artificial limitation now. 

Also - make this function return the set, and throw the necessary errors in the main control flow.

Reply:
Yes, I've tried this with @guan840912 from CFN's perspective and it returns failure. So we have to throw an error here.